numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1783,7 +1803,20 @@ unpack_bits(PyObject *input, int axis, PyObject *count_obj)
     /* setup lookup table under GIL, big endian 0..256 as bytes */
     if (unpack_init == 0) {
         npy_uint64 j;
-        npy_uint64 * unpack_lookup_64 = (npy_uint64 *)unpack_lookup;
+        npy_uint64 * unpack_lookup_64 = (npy_uint64 *)unpack_lookup_l;
+        for (j=0; j < 256; j++) {
+            npy_uint64 v = 0;
+            v |= (npy_uint64)((j &   1) ==   1);
+            v |= (npy_uint64)((j &   2) ==   2) << 8;
+            v |= (npy_uint64)((j &   4) ==   4) << 16;
+            v |= (npy_uint64)((j &   8) ==   8) << 24;
+            v |= (npy_uint64)((j &  16) ==  16) << 32;
+            v |= (npy_uint64)((j &  32) ==  32) << 40;
+            v |= (npy_uint64)((j &  64) ==  64) << 48;
+            v |= (npy_uint64)((j & 128) == 128) << 56;
+            unpack_lookup_64[j] = v;
```

Comment:
I'm confused: exactly the same `v` is constructed here as below, the only difference being the byte swap.  Can we just fill both big and little in one loop? (one the unswapped, the other the swapped `v`)?

Reply:
fixed